src/V3Sched.cpp

```diff
@@ -598,9 +595,30 @@ const TriggerKit createTriggers(AstNetlist* netlistp, AstCFunc* const initFuncp,
     AstScope* const scopeTopp = topScopep->scopep();
     FileLine* const flp = scopeTopp->fileline();
 
+    // Gather all the unique SenItems under the SenTrees
+    // List of unique SenItems used by all 'senTreeps'
+    std::vector<const AstSenItem*> senItemps;
+    // Map from SenItem to the equivalent index in 'senItemps'
+    std::unordered_map<const AstSenItem*, size_t> senItemp2Index;
+    {
+        // Set of unique SenItems
+        std::unordered_set<VNRef<const AstSenItem>> uniqueSenItemps;
```

Comment:
I'm surprised this works, I'd think the VL_LEAK checks would complain... we don't make direct usage of AstNodes anywhere else, so please use a pointer or otherwise here.

Reply:
Not sure what you think the issue is.

The point here is that we want two different instances of the same SenItem (e.g.: both representing `posedge clk`) to be considered the same. VNRef was designed to achieve exactly this:
https://github.com/verilator/verilator/blob/5a65d278511ffc7801b0387482f877fd4eda3a76/src/V3Ast.h#L3016-L3019
And we use the same thing elsewhere:
https://github.com/verilator/verilator/blob/5a65d278511ffc7801b0387482f877fd4eda3a76/src/V3SenExprBuilder.h#L39-L43

Reply:
That is:

`std::unordered_set<VNRef<const AstSenItem>>`
Is the same as
`std::unordered_set<const AstSenItem*>`
except it uses equivalence comparison (sameTree) instead of identity (pointer equality)

Reply:
Sorry, I misread, I thought the needed syntax was `VNRef<AstNode*>`, which in retrospect if that was the syntax obviously would read wrong ;)

Reply:
Yeah, the one extra difference between `std::unordered_set<const AstSenItem*>` and `std::unordered_set<VNRef<const AstSenItem>>` is that the latter can't store a null pointer, hence the syntax.